app/src/main/java/co/kaush/msusf/movies/MSMovieViewModel.kt

```diff
@@ -1,44 +0,0 @@
-package co.kaush.msusf.movies
-
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
-import androidx.lifecycle.viewModelScope
-import co.kaush.msusf.MSApp
-import co.kaush.msusf.usf.UsfVm
-import kotlinx.coroutines.CoroutineScope
-
-/**
- * For this example, a simple ViewModel would have sufficed, but in most real world examples we
- * would use an AndroidViewModel
- *
- * Our Unit tests should still be able to run given this
- */
-class MSMovieViewModel(
-    app: MSApp,
-    movieRepo: MSMovieRepository,
-    coroutineScope: CoroutineScope? = null,
-) : AndroidViewModel(app), UsfVm<MSMovieEvent, MSMovieResult, MSMovieViewState, MSMovieViewEffect> {
```

Comment:
Refactored generated ViewModel: no longer implements `UsfVm` interface, simplifying KSP logic. This change is transparent from an API user perspective. Future modifications may bring KSP generation code closer to this approach.

@CubanAzcuy was not comfortable with some of the hard-coding that was required in the ksp gen code, to enable the implementation of the interface.